frontend/packages/kubevirt-plugin/src/components/virtualization-overview/running-vms-per-template-card/RunningVmsPerTemplateCard.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { ChartDonut, ChartLabel, ChartLegend } from '@patternfly/react-charts';
+import { Card, CardBody, CardHeader, CardTitle } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { LABEL_USED_TEMPLATE_NAME } from '../../../constants';
+import { VirtualMachineModel } from '../../../models';
+import { kubevirtReferenceForModel } from '../../../models/kubevirtReferenceForModel';
+import { VMKind } from '../../../types';
+
+import './running-vms-per-template-card.scss';
+
+const getVmsPerTemplateMap = (vms, vmsLoaded) => {
+  let numVMs = 0;
+  const map = {};
```

Comment:
Following code seems to be a combined computation of `numVMs` and `map` - I'd recommend to split it up for better code readability (and to avoid using `let` if possible), for example:

```ts
const map = {};

if (vmsLoaded) {
  vms.forEach((vm) => {
    const template = vm?.metadata?.labels[LABEL_USED_TEMPLATE_NAME];
    map[template] = map.hasOwnProperty(template) ? map[template] + 1 : 1;
  });
}

const numVMs = Object.keys(map).length;
```

Reply:
Done!